// react-native-audio-api/lib/module/core/AudioBufferBaseSourceNode.js
"use strict";

import AudioParam from "./AudioParam.js";
import AudioScheduledSourceNode from "./AudioScheduledSourceNode.js";
export default class AudioBufferBaseSourceNode extends AudioScheduledSourceNode {
  constructor(context, node) {
    super(context, node);
    this.detune = new AudioParam(node.detune, context);
    this.playbackRate = new AudioParam(node.playbackRate, context);
  }
  get onPositionChanged() {
    return this.positionChangedCallback;
  }
  set onPositionChanged(callback) {
    if (!callback) {
      this.node.onPositionChanged = '0';
      this.positionChangedSubscription?.remove();
      this.positionChangedSubscription = undefined;
      this.positionChangedCallback = undefined;
      return;
    }
    this.positionChangedCallback = callback;
    this.positionChangedSubscription = this.audioEventEmitter.addAudioEventListener('positionChanged', callback);
    this.node.onPositionChanged = this.positionChangedSubscription.subscriptionId;
  }
  get onPositionChangedInterval() {
    return this.node.onPositionChangedInterval;
  }
  set onPositionChangedInterval(value) {
    this.node.onPositionChangedInterval = value;
  }
}
